### ISSUE TITLE
Optimization: remove array_replace_recursive calls in setConfig in module

### ISSUE THREAD

**Issue by derekisbusy** [communityii/yii2-user#49]
Instead of using array_replace_recursive to set all the defaults every time the module loads should switch the code to use the `Module::getSetting()` and related functions ie. `getRegistrationSetting()`

So in the controller you would use
```php
$this->module->getSocialSetting('enabled');
```

This way the defaults are only loaded when needed.

Or we could remove the getSocialSettings and variable from the module and make components out of all the settings and attach them to the user component.

```php
Yii::$app->user->social->enabled;
```

This way components and defaults are only set when needed.

**Comment by kartik-v** [comment]
Sure go ahead. I have pushed updates for social auth.

**Comment by kartik-v** [comment]
Sure go ahead. I have pushed updates for social auth in #40.

Thinking ahead... this now needs some test cases (codeception or similar) for ensuring changes do not break.

**Comment by derekisbusy** [comment]
The social auth is broke right now.

**Comment by derekisbusy** [comment]
Okay never mind I see you changed it.

**Comment by kartik-v** [comment]
Yes social auth has been redone by me. It should work - need to add all the Auth clients (have done for Google, Facebook and Twitter).

Following needs to be noted and done:

- Removing `array_replace_recursive` would adversely impact many code areas including defaults that are used in views, widgets as well. Need to be really careful in doing this.
- Testing is important from here. Would suggest if you can help build test cases in a new folder `tests` to test each of these. I can help add to the same as well.
